docs/source/hooks.rst

```diff
@@ -0,0 +1,135 @@
+.. _hooks:
+
+=====
+Hooks
+=====
+
+.. _hooks_introduction:
+
+Introduction
+============
+
+One of the goals of the :py:class:`~signac` framework is to make it easy to track the provenance of research data
+and to ensure its reproducibility.
+Hooks make it possible to track these state changes in a project.
+In general, a hook is a function that is called at a specific point relative to the execution of an operation.
+
+Hooks execute code adjacent to key steps of an operation,
+such as when it begins, finishes, or fails. These events are
+called triggers.
+Hooks help track where an execution's data originated, and which operations were applied to the data.
```

Comment:
I prefer "track an execution's data" since it's more specific -- it refers to which pieces of code and how the code looks at the time the data was generated, while "track a job's data" is more ambiguous since it lacks the temporal implication. Is "an execution" to jargon-y?